llvm/lib/Target/AMDGPU/AMDGPURewriteAGPRCopyMFMA.cpp

```diff
@@ -208,8 +218,19 @@ bool AMDGPURewriteAGPRCopyMFMAImpl::run(MachineFunction &MF) const {
 
       CopySrcMI->setDesc(TII.get(AGPROp));
 
-      // TODO: Is replacing too aggressive, fixup these instructions only?
-      MRI.replaceRegWith(CopySrcReg, VReg);
+      // Perform replacement of the register, rewriting the rewritable uses.
+      for (MachineInstr &UseMI :
```

Comment:
Is `UseMI` here misleading as `reg_instructions` will also include defs?

Reply:
I think it's a bit confusing too. The comment above also only mentions uses as being rewritten whereas defs will also have their register substituted.